gradle.template.properties

```diff
@@ -8,3 +8,7 @@ org.gradle.daemon=false
 # e.g org.gradle.java.home=/Library/Java/JavaVirtualMachines/jdk1.8.0_144.jdk/Contents/Home
 # e.g org.gradle.java.home=C:/Program Files/Java/jdk1.8.0_144
 # org.gradle.java.home=
+
+# Use this property to have the recommended language setting for tests
+# It is used to prevent local language settings causing the tests to fail
```

Comment:
It doesn't appear to work for me in IntelliJ and it won't quite make sense because running tests in IntelliJ itself would not use the Gradle JVM.

How did you manage to make this work in IntelliJ because there must be something I'm missing here.

Reply:
Can you confirm this?

Reply:
If I am not mistaken, IntelliJ only supports English so this shouldn't be a problem itself.
On making my locale German and disabling the gradle JVM argument, the tests still pass when run through IntelliJ.


May I know what steps you took which didn't let this work on IntelliJ?

Reply:
For me no matter what locale, tests always pass when run through IntelliJ IDE. I assumed it was because the IDE is only configured for English and its not possible to change locale of IDE only. Correct me if I am wrong

Reply:
It is not possible to change the locale of the IDE but in the case of macOS if the user only has one language specified the JVM does not set `user.language.format`. So if my language is German, `user.language` will be set to `de` and the test will still fail in IntelliJ.

However, if I have more than one language in my operating system, `user.language.format` will be only be set to `en` in IntelliJ since macOS High Sierra (http://www.openradar.me/34581827).

I didn't check how this works on WIndows though.

Reply:
You were right. I didn't think of this case. 

On researching, I found that to get around this, we can set the `user.country` and `user.language` properties if IntelliJ. But currently, this can only be done by changing manually through `idea.exe.vmoptions` file or by passing through JVM arguments.

Better of this options is to set default JVM arguments, but I am not sure how to do it for any Run. Currently on CITests use `-ea -Xss2m` as default parameters. Do you know how can I make certain arguments as default when anything is run for this project?

Reply:
For IntelliJ we can go `Run...` -> `Edit Configurations...` and set the defaults for `TestNG`. 

We need to do this for Eclipse though as well.

Maybe it's easier if we just set the default locale in Java since the App Engine server will always be running in the locale `en_US` anyway.

Reply:
> For IntelliJ we can go Run... -> Edit Configurations... and set the defaults for TestNG.

Can we not do this in a similar manner of how its done for `CI_Tests` in runConfigurations in .idea folder?

> just set the default locale in Java

Through `Locale.setLocale()` or by setting `-Duser.country=US -Duser.language=en`(not sure how to do this) as the default JVM arguments?

Reply:
> Can we not do this in a similar manner of how its done for CI_Tests in runConfigurations in .idea folder?

I'm afraid it seems that's the best we can do. At the moment we don't need to but perhaps if we ever need to we can modify `workspace.xml`. But it's probably easier to tell the user to set their defaults.

Share run configuration defaults: https://youtrack.jetbrains.com/issue/IDEA-65915

> Through Locale.setLocale()

I meant this.